pyhon/appliance.py

```diff
@@ -297,10 +307,6 @@ def sync_parameter(self, main: Parameter, target: Parameter) -> None:
             target.max = main.max
             target.min = main.min
             target.step = main.step
-        elif isinstance(target, HonParameterRange):
-            target.max = int(main.value)
-            target.min = int(main.value)
-            target.step = 1
```

Comment:
This is important for other functions and will introduce problems for other appliances if we remove it. 
Please use `startProgram.program` to switch the mode instead of `settings.machMode`.

Reply:
Wait. Looks like my explanation is not so good.
So.
I use command 'startProgram' for turn on water heater. I have no any other ways to turn on appliance.
Next. startProgram has 3 params:
onOffStatus (mandatory, fixed) 
machMode (optional, fixed)
tempSel (optional, fixed)

However, real settings.machMode is range, and has 1,2 and 3 modes (eco, max, bps). And it's correct type of this parameter. 
I don't know why manufacturer implement parameter 'startProgram.machMode' as fixed. But, we have what we have. 

As result. I call start program - >sync command replace range of 'settings.machMode' with only one value. It's incorrect. More over. I didn't touch sync if base and target parameter are range. I removed sync code only for case if base parameter is fixed and target is range. That's it. 
Let me know of you need more examples.

Reply:
@Andre0512
Any comments or suggestions from your side?

Reply:
@Andre0512 hey man

Reply:
Hi, sorry you had to wait so long! 
So I can`t merge this unfortunately if it removes line 301-303, this will break functionalities of some appliances...

If you change `startProgram.program` to 'eco' then `machMode` is 1, if you change `startProgram.program` to 'max' then `machMode` is 2 and so on. So you don`t need to change `machMode` itself at any time, you can work with changing `startProgram.program` and it`s ok that `machMode` is fixed, it will be updated with the changing program. This is how I implemented it for other appliances.
Is it understandable what I mean and can you try it? :slightly_smiling_face:

Reply:
@Andre0512 Thanks. I understood the concept. It was not clear for me at the start.
So, I rewrite my fork of hOn according to your explanation. All work. I revert back `sync_command`
Should be fine now.